sandbox/lua/decoders/syslog.lua

```diff
@@ -0,0 +1,182 @@
+    | name:"postfix_delay" value:0.05
+    | name:"postfix_delay_before_qmgr" value:0.05
+    | name:"postfix_delay_in_qmgr" value:0
+    | name:"postfix_delay_conn_setup" value:0
+    | name:"postfix_delay_transmission" value:0
+    | name:"postfix_dsn" value:"2.0.0"
+    | name:"postfix_status" value:"sent"
+--]]
+
+local l = require 'lpeg'
+local syslog = require 'syslog'
+local syslog_message = require 'syslog_message'
+local postfix = require 'postfix'
+
+local rsyslog_template = read_config('rsyslog_template')
+local msg_type = read_config('type')
+local hostname_keep = read_config('hostname_keep')
+
+local msg = {
+    Timestamp   = nil,
+    Type        = msg_type,
+    Hostname    = nil,
+    Payload     = nil,
+    Pid         = nil,
+    Severity    = nil,
+    Fields      = nil
+}
+
+local rsyslog_grammar = nil
+if rsyslog_template then
+    rsyslog_grammar = syslog.build_rsyslog_grammar(rsyslog_template)
+    if not rsyslog_grammar then
+        error('Unable to parse rsyslog template')
+    end
+end
+
+local wildcard_grammars = syslog_message.get_wildcard_grammar()
+
+function process_message ()
+    local log = read_message('Payload')
+    local fields = nil
+    if rsyslog_grammar then
+        fields = rsyslog_grammar:match(log)
+        if not fields then return -1 end
+        if fields.timestamp then
+            msg.Timestamp = fields.timestamp
+            fields.timestamp = nil
+        end
+
+        if fields.pri then
+            msg.Severity = fields.pri.severity
+            fields.syslogfacility = fields.pri.facility
+            fields.pri = nil
+        else
+            msg.Severity = fields.syslogseverity or fields['syslogseverity-text']
+            or fields.syslogpriority or fields['syslogpriority-text']
+
+            fields.syslogseverity = nil
+            fields['syslogseverity-text'] = nil
+            fields.syslogpriority = nil
+            fields['syslogpriority-text'] = nil
+        end
+
+        if fields.syslogtag then
+            fields.programname = fields.syslogtag.programname
+            msg.Pid = fields.syslogtag.pid
+            fields.syslogtag = nil
+        end
+
+        if not hostname_keep then
+            msg.Hostname = fields.hostname or fields.source
+            fields.hostname = nil
+            fields.source = nil
+        end
+
+        msg.Payload = fields.msg
+        fields.msg = nil
+
+    else
+        msg = {}
+        msg.Uuid = read_message('Uuid')
+        if msg_type then
+            msg.Type = msg_type
+        else
+            msg.Type = read_message('Type')
+        end
+        msg.Logger = read_message('Logger')
+        msg.Payload = read_message('Payload')
+        msg.EnvVersion = read_message('EnvVersion')
+        msg.Hostname = read_message('Hostname')
+        msg.Timestamp = read_message('Timestamp')
+        msg.Severity = read_message('Severity')
+        msg.Pid = read_message('Pid')
+        fields = read_message('Fields')
```

Comment:
@trink. Any alternative?

I expect having fields like _SYSTEMD_CGROUP=/user.slice/user-1000.slice/session-1.scope here.

If this is not possible, I will make rsyslog_template mandatory for now and we'll see later on for systemd-journal integration